File: leet/two_pointer/container_with_most_water_11.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;
        int maxAmount = 0;

        while (left < right) {
            int bottom = right - left;
            int tall = Math.min(height[left], height[right]);
            maxAmount = Math.max(maxAmount, bottom * tall);
            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }
        return maxAmount;
    }
}

File: leetcode_75/two_pointer/container_with_most_water_11.py
from typing import List

class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right, max_amount = 0, len(height)-1, 0

        while left<right:
            bottom = right-left
            tall = min(height[left],height[right])
            max_amount = max(max_amount,bottom*tall)

            if height[left] < height[right]:
                left+=1
            else:
                right-=1
        return max_amount